### PR TITLE
feat(io): unify model load/save paths, promote dtype precision, add stage-parity tooling

### PR DESCRIPTION
**Pull request by Gelu-Nita**
## Summary

This branch consolidates a large number of improvements accumulated since PR #41:

- Unified model I/O: `_sav_convert.py` replaces `tests/build_h5_from_sav.py`; public API through `pyampp.io`
- `pyampp/io/model.py`: expanded canonical load/save with robust metadata path resolution
- `pyampp/gxbox/gx_fov2box.py`: major refactor with `--reproject-algorithm` flag, legacy NLFFF ctypes compatibility patch, improved stage-resume and entry-box handling
- `pyampp/io/_sav_convert.py`: corona and base fields promoted from float32 to **float64**; line indices normalised to int32
- New CLI utilities: `export_model.py`, `show_model_metadata.py`, `export_model_metadata.py`; renamed from h5thin family
- Docs: geometry_api, model_io, viewers, gx_fov2box_usage all updated

## real_data_checks additions

Four new analysis scripts documenting the OBS→NONE parity investigation:

| Script | Purpose |
|---|---|
| `check_idl2py_stage_parity.py` | multi-stage IDL-vs-Python parity driver; now supports `--reproject-algorithm` |
| `rank_obsnone_reproject_scan.py` | rank adaptive/exact/interpolation reprojection against IDL NONE reference |
| `plot_obsnone_relative_error_panel.py` | vector-magnitude relative error panel PNG |
| `plot_obsnone_component_metric_panels.py` | per-component abs-residual and sym-diff panels |
| `export_none_base_maps_to_fits.py` | export NONE base maps to IDL-readable FITS MEF |

Full methodology and findings documented in `real_data_checks/README.md`.

## Key analysis findings

- **adaptive > exact > interpolation** for OBS→NONE base-map fidelity against IDL reference
- **Boundary injection test**: pyAMPP-generated NONE boundary conditions injected back into IDL run all downstream stages (POT→NAS→GEN→CHR) successfully; NAS corona mean_rel improves from 0.36 (original IDL) to 0.009 (with pyAMPP NONE)
- **OBS→NONE gap isolated**: the remaining parity gap is confirmed to live exclusively in the OBS→NONE reprojection/boundary step; all downstream stages show correct behaviour
- Designated as **dedicated post-merge investigation** in coordination with Alexey Stupishin (GX library author)

## dtype fix (breaking for downstream consumers of HDF5 canonical format)

`base/bx`, `base/by`, `base/bz`, `base/ic` previously written as float32 in some code paths; now always float64 to match IDL source precision.

## Tests

- `test_fov2box_time_anchor.py` — extended significantly
- `test_model_io_canonical_contract.py` — new suite
- `test_runtime_model_entrypoints.py` — new
- `test_show_model_metadata_cli.py` — new
- `test_h5tree_cli.py` — replaces `test_h5thin_cli.py`

## Breaking changes / renames

- `pyampp/util/h5thin.py` → `show_model_metadata.py`
- `pyampp/util/h5thin_export.py` → `export_model_metadata.py`
- `pyampp/tests/build_h5_from_sav.py` → `pyampp/io/_sav_convert.py` (moved to production package)
- `reports/parity_review/PR_0_2_0_CHECKLIST.md` removed (superseded by real_data_checks/README.md)

### CODE REVIEW

**Review by Gelu-Nita — COMMENTED** [review]
## Automated pre-merge review

### Summary
Branch `feat/io-only-model-load-paths` is ready for upstream review. 50 files changed, +8198/−1696 lines. All changes are functional; documentation is complete.

### ✅ Passing checks

**Security**
- All `subprocess.run` calls use list-based arguments (no `shell=True`); `shlex.split` used for command construction
- No `eval`, `exec`, or `os.system` in the diff
- No secrets or credentials introduced

**Precision / dtype correctness**
- `pyampp/io/_sav_convert.py`: corona and base magnetic field arrays now explicitly cast to `np.float64` (previously float32 in some paths); line indices normalised to int32 — consistent with IDL source precision
- All new FITS export code writes float64 (BITPIX=-64) matching the HDF5 source

**Subprocess / IPC**
- `gx_fov2box.py` legacy NLFFF ctypes patch uses `np.ctypeslib.ndpointer` with explicit dtype/flags — no unsafe pointer casts

**New scripts in real_data_checks/**
- All 4 new analysis scripts are pure-Python, read-only operations on HDF5/JSON inputs; no side effects beyond writing PNG/JSON/FITS output files to user-specified paths
- No hardcoded absolute paths in importable library code (local paths only in `real_data_checks/` scripts, documented as such)

### ⚠️ Items for reviewer attention

1. **dtype promotion is a format change**: `base/bx, by, bz, ic` and `corona/bx, by, bz` are now always float64 in freshly exported HDF5 files. Any downstream code reading these fields and comparing against previously cached float32 HDF5 files will see a format difference. This is intentional and documented in the PR body and README.

2. **Renamed utilities**: `h5thin.py→show_model_metadata.py`, `h5thin_export.py→export_model_metadata.py`. If any user scripts reference the old names they will break. Entry-point names in `pyproject.toml` should be verified to match.

3. **`_sav_convert.py` is now production code** (moved from `pyampp/tests/` to `pyampp/io/`). The leading underscore signals it is not part of the public API, but reviewers should confirm the module is not accidentally importable by users expecting stability.

4. **OBS→NONE gap not addressed in this PR**: the reprojection algorithm analysis (adaptive > exact > interpolation) and the boundary injection test confirm the gap exists and is localised. This PR only adds the tooling and documents the findings. A dedicated follow-up branch is planned.

### CI
2 pending checks (py3.10, py3.12). No failures at time of review.

File: real_data_checks/export_none_base_maps_to_fits.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+"""Export base-layer maps from a pyAMPP NONE H5 file to a FITS file readable in IDL.
+
+Output: multi-extension FITS (MEF).
+  HDU 0  – empty primary with run metadata
+  HDU 1  – BX        (base/bx,  float32,  Gauss)
+  HDU 2  – BY        (base/by,  float32,  Gauss)
+  HDU 3  – BZ        (base/bz,  float32,  Gauss)
+  HDU 4  – IC        (base/ic,  float32,  continuum intensity, normalised)
+  HDU 5  – CHROMO_MASK  (base/chromo_mask, int16)
```
[the remark this answers is not in the document]
Comment (Gelu-Nita) [reply real_data_checks/export_none_base_maps_to_fits.py]: Fixed in commit 825f800: updated  docstring to  for BX/BY/BZ/IC and  for CHROMO_MASK, matching the actual tuple values.

File: pyampp/io/model.py [hunk 2]
```diff
@@ -33,11 +25,146 @@
...
+        except Exception:
+            pass
+    # Otherwise, compute and inject
+    new_contract = complete_geometry_contract(model_dict, strict=strict)
+    if new_contract is not None:
+        metadata["geometry_contract"] = new_contract
+        model_dict["metadata"] = metadata
 from pyampp.gxbox.boxutils import (
     normalize_observer_metadata,
 )
```
[the remark this answers is not in the document]
Comment (Gelu-Nita) [reply pyampp/io/model.py]: Fixed in commit 825f800: moved `ensure_geometry_contract_in_metadata` below all import statements. The function is intentionally public (used by downstream consumers that need to ensure contract presence before passing a model dict into pipeline stages) so the name is kept without a leading underscore.

File: real_data_checks/export_none_base_maps_to_fits.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Gelu-Nita) [reply real_data_checks/export_none_base_maps_to_fits.py]: Fixed in commit 825f800: updated the `_COMPONENTS` docstring to `float64` for BX/BY/BZ/IC and `int32` for CHROMO_MASK, matching the actual tuple values. (Re-posting — previous reply had backticks stripped by the shell.)
